.github/workflows/cloudflare-preview-forks.yml

```diff
@@ -0,0 +1,420 @@
+            else
+              echo "::error::Could not determine output directory in $PWD. Set repo variable PAGES_OUTPUT_DIR."
+              exit 1
+            fi
+          fi
+          echo "Using output dir: $OUTDIR"
+          echo "outdir=$OUTDIR" >> "$GITHUB_OUTPUT"
+
+      - name: Package built site
+        run: |
+          mkdir -p artifact
+          SRC="${{ steps.workdir.outputs.workdir }}/${{ steps.outdir.outputs.outdir }}"
+          echo "Copying from: $SRC"
+          cp -a "$SRC"/. artifact/
+          echo "Packaged $(find artifact -type f | wc -l) files."
+
+      - name: Upload built artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: site-dist
+          path: artifact
+          if-no-files-found: error
+          retention-days: 7
+
+  deploy:
+    name: Deploy preview to Cloudflare Pages
+    if: ${{ github.event.pull_request.head.repo.fork == true && github.event.action != 'closed' }}
+    runs-on: ubuntu-latest
+    needs: build
+    permissions:
+      contents: read
+      pull-requests: write
+      issues: write
+    steps:
+      - name: Load 1password secret(s)
+        uses: 1password/load-secrets-action@v3
+        with:
+          export-env: true
+        env:
+          OP_SERVICE_ACCOUNT_TOKEN: "${{ secrets.TESTS_SERVICE_ACCOUNT_TOKEN }}"
+          CF_API_TOKEN: "op://test-secrets/CF_API_TOKEN/credential"
+
+      - name: Check required secrets
+        env:
+          CF_ACCOUNT_ID: ${{ vars.CF_ACCOUNT_ID }}
+          CF_PAGES_PROJECT: ${{ vars.CF_PAGES_PROJECT }}
+        run: |
+          missing=0
+          for v in CF_API_TOKEN CF_ACCOUNT_ID CF_PAGES_PROJECT; do
+            if [ -z "${!v}" ]; then
+              echo "::error::Missing repository secret '$v'."
+              missing=1
+            fi
+          done
+          if [ "$missing" -ne 0 ]; then
+            echo "Set CF_API_TOKEN, CF_ACCOUNT_ID, CF_PAGES_PROJECT in repo settings. CF_PAGES_PROJECT must be a Cloudflare Pages Direct Upload project."
+            exit 1
+          fi
+
+      - name: Download built artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: site-dist
+          path: site-dist
+
+      - name: Publish to Cloudflare Pages (preview)
+        id: pages
+        uses: cloudflare/pages-action@v1
```

Comment:
This action is deprecated https://github.com/cloudflare/pages-action
Suggested replacement https://github.com/cloudflare/wrangler-action

Related:

- https://github.com/cloudflare/wrangler-action/issues/301